src/ImageSharp/Image.cs

```diff
@@ -67,7 +67,7 @@ protected Image(Configuration configuration, PixelTypeInfo pixelType, ImageMetad
         public int Height => this.size.Height;
 
         /// <inheritdoc/>
-        public ImageMetadata Metadata { get; }
+        public ImageMetadata Metadata { get; internal set; }
```

Comment:
Should there to be a public API for setting this somewhere? With the new setter being `internal`, a third party image decoder can't properly re-use an `Image` as it can't change the metadata.

Reply:
You set individual properties on the metadata not the metadata itself. 

See [ImageMetadata](https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.Metadata.ImageMetadata.html#properties)

Reply:
I see, that makes sense. Seeing different `*Metadata` classes had me confused for a moment, not noticing that `ImageMetadata` is sealed.